src/plugins/passwd/passwd.c

```diff
@@ -129,24 +129,26 @@ int elektraPasswdGet (Plugin * handle ELEKTRA_UNUSED, KeySet * returned ELEKTRA_
 	else
 		index = UID;
 	struct passwd * pwd;
-#if HAS_FGETPWENT
+#if defined(_GNU_SOURCE)
 	FILE * pwfile = fopen (keyString (parentKey), "r");
 	if (!pwfile)
 	{
 		ELEKTRA_SET_ERRORF (110, parentKey, "Failed to open configuration file %s\n", keyString (parentKey));
 		return -1;
 	}
 	while ((pwd = fgetpwent (pwfile)) != NULL)
-#else
+#elif defined(_X_OPEN_SOURCE)
 	while ((pwd = getpwent ()) != NULL)
+#else
+#error Need either _GNU_SOURCE or _X_OPEN_SOURCE
```

Comment:
please improve error message: doesn't this case also mean that there is a bug in CMake code?

Reply:
yes this should not be reached. Should I replace it with something else?

Reply:
Simply say that CMake should make sure that this is not reached and that this is a bug in Elektra. (ideally with URL to issues.libelektra.org)

Reply:
aa27585